index.js

```diff
@@ -83,50 +78,78 @@ Collector.prototype.computePointCountToSend = function(pointSizes, upperBound) {
   return Math.min(index, pointSizes.length); // But not if there were no points at all.
 };
 
-Collector.prototype.flush = function() {
-    var self = this;
-
-    if (!self._points || self._points.length === 0) {
+function flushSeries(collector, seriesName, points) {
+    if (!points || points.length === 0) {
         return;
     }
 
     // only send N points at a time to avoid making requests too large
-    // TODO what if we are backed up?
-
-    var spliceIndex = self.computePointCountToSend(self._points.map(function (point) {
-      return JSON.stringify(point).length;
-    }), MTU_SIZE);
-    var points = self._points.splice(0, spliceIndex);
-    var opt = { precision: self._time_precision };
+    var spliceIndex;
+    if (collector._client.protocol == 'udp:') {
+      spliceIndex = collector.computePointCountToSend(collector._points.map(function (point) {
+        return JSON.stringify(point).length;
+      }), MTU_SIZE);
+    } else {
+      spliceIndex = 50;
+    }
+    var batch = points.splice(0, spliceIndex);
+    var opt = { precision: collector._time_precision };
 
-    self._client.writePoints(self._series_name, points, opt, function(err) {
+    collector._client.writePoints(seriesName, batch, opt, function(err) {
         if (err) {
             // TODO if error put points back to send again?
             return self.emit('error', err);
         }
 
         // there are more points to flush out
-        if (self._points.length >0) {
-            setImmediate(self.flush.bind(self));
+        if (points.length > 0) {
+            flushSeries(collector, seriesName, points);
         }
     });
+}
+
+Collector.prototype.flush = function() {
+    var self = this;
+
+    Object.keys(self._series).forEach(function(key) {
+        var series = self._series[key];
+        delete self._series[key];
+
+        flushSeries(self, key, series);
+    });
 };
 
+function getSeries(collector, name, reset) {
+    var series = collector._series[name];
+
+    if(!series) {
+        series = [];
+        collector._series[name] = series;
+    }
+
+    return series;
+}
+
 // collect a data point (or object)
 // @param [Object] value the data
 // @param [Object] tags the tags (optional)
-Collector.prototype.collect = function(value, tags) {
+Collector.prototype.collect = function(seriesName, value, tags) {
     var self = this;
 
-    // disabled (due to no URL)
-    if (!self._points) {
-        return;
-    }
-
-    self._points.push([value, tags]);
     if (self._instant_flush) {
-        self.flush();
+        flushSeries(self, seriesName, [[value, tags]]);
+    } else {
+        var series = getSeries(self, seriesName);
+
```

Comment:
Nit: no need for the blank space, the two lines seem pretty related to each other.